docs/design/reserve-action.md

```diff
@@ -0,0 +1,261 @@
+and `2C2G` was released from node. Because of that the idle resource of node can not satisfy requirement of pod `pod-4`, `pod-4`
+was skipped. The `pod-6` and `pod-7` were bound to node and begin to run. In `Time6` all pods except pod `pod-4` were completed
+and all resources of node were release. The scheduling condition of pod `pod-4` is satisfied, `pod-4` was bound to node and 
+begin to run. In the above scenario, the `pod-4` can not be scheduled in time because a large number of small resource requirement
+pods occupied the node resources.
+ 
+In order to solve such problems, the reserve action is introduced into the scheduling policy. The `reserve` action can 
+prevents the large resource requirement task from being starved by reserving node resources for it and not allowing the other 
+tasks to be scheduled on the node.
+
+![reserve-resource-for-pod](../../docs/images/reserve-resource-for-pod.png)
+
+Different from the original scheduling logic, in `Time4`, although there are `2C2G` idle resources on the node, it does not
+schedule new tasks to the nodes, but reserves these resources for `pod-4`. In `Time5` `pod-6` and `pod-7` remain `Pending`. 
+In `Time 6`, `pod-5` completed and all resources on node were released. The scheduling condition of `pod-4` is satisfied, `pod-4`
+is bound to the node and starts `Running`.
+
+## Function Detail 
+
+### Explanation of terms
+
+#### Job Ready
+
+If the running condition of a job is satisfied, then the job is in the `Job Ready` state, and after scheduling, the job can be in the 
+`Job Running` state. The running condition of a job may include restrictions such as `gang-scheduler` and so on.
+
+#### Job Condition Ready
+
+If the running condition of a job is satisfied under certain conditions, then the state of the job can be called `Conditional Running`.
+When after the node reserving resources for the tasks under the job, the job can reach the running condition, then the state of 
+the job can be called `Condition Ready`.
+
+#### Job starving
+
+The starving job needs to meet the following conditions
+
+* The creation time of job is greater than the set starving job threshold time
+* Job state is not `Job Ready`
+* Job state is not `Job Condition Ready`
+
+If a job has been created for a long time, but it is still not `running`, it is at risk of starving. If nodes under the cluster
+have reserved resource for the starving job, then there is no need to reserve resources for it again.
+
+#### ForceAllocated
+
+If a node tries to reserve resources for the task under the job, when the node has not reserved resources for other tasks, and 
+the idle resources of the node are larger than the requirement resources of the task, this task can be directly allocated to 
+this node with `ForceAllocated` state.  The `ForceAllocated` state is to avoid subsequent `backfill` action to preempt pod
```

Comment:
Could `preempt` action preempt `ForceAllocated` and `FakeAllocated` pods?

Reply:
The `preempt` action can only evict `running` tasks, will not evict `FakeAllocated` or `ForceAllocated` tasks

Reply:
It seems not very reasonable that high priority task could not get resource?

Reply:
Yes, in the `reserve` process, when trying to reserving resource for a starving task, we need to ensure that the tasks whose priority is higher than the starving task will be prioritized to schedule and prioritized to get resources. I will make an update here.

Reply:
Added a new chapter `Reserve ligic` to introduce the logic of processing jobs who have higher priority than the starving job.